YourEasyRentTest/Controllers/ProductControllerTest.cs

```diff
@@ -0,0 +1,264 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Moq;
+using FluentAssertions;
+using YourEasyRent.Entities;
+using YourEasyRent.DataBase.Interfaces;
+using YourEasyRent.Controllers;
+using YourEasyRent.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace YourEasyRentTest.Controllers
+{
+    public class ProductControllerTest
+    {
+        [Fact]
+        public async Task GetProducts_Return_AllProducts()
+        {
+            // arrange
+            var products = new List<Product>{new Product { Id = "test", Name = "Product 1" }, new Product { Id = "test", Name = "Product 2" } };       
+            var productRepoMock = new Mock<IProductRepository>(); 
+            productRepoMock.Setup(repo => repo.GetProducts()).ReturnsAsync(products);
+            var controller = new ProductController(productRepoMock.Object); // Создала экземпляр контроллера ProductsController, передавая фейковый repositoryMock в конструктор.
+
+            // act
+            var result = await controller.GetProducts();
+
+            // assert
+            result.Should().BeOfType<ActionResult<IEnumerable<Product>>>()
```

Comment:
It would be nice to also check that products data like an "id" or "Name" fields are mapped correctly.